cmd/sqlcmd/main_test.go

```diff
@@ -162,8 +162,8 @@ func TestUnicodeOutput(t *testing.T) {
 	}
 	vars := sqlcmd.InitializeVariables(!args.DisableCmdAndWarn)
 	setVars(vars, &args)
-
-	exitCode, err := run(vars, &args)
+	sqlcmdConsole := getSqlcmdConsole(args)
```

Comment:
let's go with a model that doesn't require so many changes.
Just define a type like
```go

type allocConsole func(historyFile string) sqlcmd.Console

```
And create a field whose default assignment is console.NewConsole. The test can override it by assigning it a new func.

Reply:
The unit test now no more relies on the console allocator function hence have removed it.
The unit test will validate console initialization and conditions for password prompt based on the parameters supplied.